src/agent/agent_info/src/agent_info.cpp

```diff
@@ -80,3 +83,33 @@ void AgentInfo::SetGroups(const std::vector<std::string>& groupList)
     agentInfoPersistance.SetGroups(groupList);
     m_groups = groupList;
 }
+
+std::string AgentInfo::CreateKey()
+{
+    static constexpr size_t key_length = 32;
+    constexpr std::string_view charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
+
+    boost::random::mt19937 generator {boost::random::random_device {}()};
+    boost::random::uniform_int_distribution<size_t> distribution(0, charset.size() - 1);
+
+    std::string key;
+    for (size_t i = 0; i < key_length; ++i)
+    {
+        key += charset[distribution(generator)];
+    }
+
+    return key;
+}
+
+bool AgentInfo::ValidateKey(const std::string& key)
+{
+    static constexpr size_t key_length = 32;
```

Comment:
We have this two times, can we have just one globally for the class?

Reply:
Now it's a `constexpr` within an anonymous namespace in the class implementation file.